### PR TITLE
hyprcursor: init at 0.1.4

### LINKED ISSUE #294395: Package request: `hyprcursor`
**Project description**

The hyprland cursor format, library and utilities.

**Metadata**

* homepage URL: https://wiki.hyprland.org/Hypr-Ecosystem/hyprcursor/
* source URL: https://github.com/hyprwm/hyprcursor
* license: BSD3
* platforms: unix, linux, darwin, ...

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by iynaix**
## Description of changes

The [hyprland cursor format](https://blog.vaxry.net/articles/2024-cursors), library and utilities.

fixes: #294395 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by iynaix** [comment]
Added the `dev` and `lib` outputs.

File: pkgs/by-name/hy/hyprcursor/package.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+, pkg-config
+, cairo
+, hyprlang
+, librsvg
+, libzip
+, nix-update-script
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hyprcursor";
+  version = "0.1.2";
```
Comment (GaetanLepage) [inline pkgs/by-name/hy/hyprcursor/package.nix:14]:
```suggestion
  version = "0.1.3";
```

File: pkgs/by-name/hy/hyprcursor/package.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hyprcursor";
+    rev = "v${finalAttrs.version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/hy/hyprcursor/package.nix:19]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/by-name/hy/hyprcursor/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    hash = "sha256-Xla4+vGnHLLpZ+8t80Rx5CoxMdIil2cHnbp9aJvL/Ts=";
```
Comment (GaetanLepage) [inline pkgs/by-name/hy/hyprcursor/package.nix:20]:
```suggestion
    hash = "sha256-41bRajBmrckIZyylzu+nFWOPu1ic/ixQE5tOgDgo0DQ=";
```

File: pkgs/by-name/hy/hyprcursor/package.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    "dev"
+    "lib"
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://github.com/hyprwm/hyprcursor";
+    description = "The hyprland cursor format, library and utilities";
+    license = lib.licenses.bsd3;
```
Comment (GaetanLepage) [inline pkgs/by-name/hy/hyprcursor/package.nix:46]:
```suggestion
    description = "The hyprland cursor format, library and utilities";
    changelog = "https://github.com/hyprwm/hyprcursor/releases/tag/v${finalAttrs.version}";
    license = lib.licenses.bsd3;
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/hy/hyprcursor/package.nix (hunk 1 above)
Comment (iynaix) [reply pkgs/by-name/hy/hyprcursor/package.nix]: Updated to 0.1.4

File: pkgs/by-name/hy/hyprcursor/package.nix (hunk 3 above)
Comment (iynaix) [reply pkgs/by-name/hy/hyprcursor/package.nix]: Updated to 0.1.4

File: pkgs/by-name/hy/hyprcursor/package.nix (hunk 4 above)
Comment (iynaix) [reply pkgs/by-name/hy/hyprcursor/package.nix]: Ok, made the change

File: pkgs/by-name/hy/hyprcursor/package.nix (hunk 2 above)
Comment (iynaix) [reply pkgs/by-name/hy/hyprcursor/package.nix]: Ok, made the change

**Review by GaetanLepage — APPROVED** [review]
Thanks !
LGTM

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 294648` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>hyprcursor</li>
    <li>hyprcursor.dev</li>
    <li>hyprcursor.lib</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 294648` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>hyprcursor</li>
    <li>hyprcursor.dev</li>
    <li>hyprcursor.lib</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 294648` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Review by drupol — APPROVED** [review]

**Pull request merged**